core/lib/dal/src/tee_proof_generation_dal.rs

```diff
@@ -126,53 +134,40 @@ impl TeeProofGenerationDal<'_, '_> {
 
     pub async fn insert_tee_proof_generation_job(
         &mut self,
-        block_number: L1BatchNumber,
+        batch_number: L1BatchNumber,
+        tee_type: TeeType,
     ) -> DalResult<()> {
-        let block_number = i64::from(block_number.0);
-        sqlx::query!(
+        let batch_number = i64::from(batch_number.0);
+        let query = sqlx::query!(
             r#"
             INSERT INTO
-                tee_proof_generation_details (l1_batch_number, status, created_at, updated_at)
+                tee_proof_generation_details (l1_batch_number, tee_type, status, created_at, updated_at)
             VALUES
-                ($1, 'ready_to_be_proven', NOW(), NOW())
-            ON CONFLICT (l1_batch_number) DO NOTHING
+                ($1, $2, 'ready_to_be_proven', NOW(), NOW())
             "#,
-            block_number,
-        )
-        .instrument("create_tee_proof_generation_details")
-        .with_arg("l1_batch_number", &block_number)
-        .report_latency()
-        .execute(self.storage)
-        .await?;
+            batch_number,
+            tee_type.to_string(),
+        );
+        let instrumentation = Instrumented::new("insert_tee_proof_generation_job")
+            .with_arg("l1_batch_number", &batch_number)
+            .with_arg("tee_type", &tee_type);
+        let result = instrumentation
+            .clone()
+            .with(query)
+            .execute(self.storage)
+            .await?;
+        if result.rows_affected() == 0 {
+            let err = instrumentation.constraint_error(anyhow::anyhow!(
+                "Unable to insert TEE proof for batch number {}, TEE type {}",
+                batch_number,
+                tee_type
+            ));
+            return Err(err);
+        }
```

Comment:
It looks like the query was idempotent before, but right now it disallows inserting the same job twice.
Moreover, without `ON CONFLICT` can it even be that `rows_affected` is 0? I'd assume that on conflict the query will fail, and otherwise we will always insert a single row.
Is there a reason for that?

Reply:
[Fixed](https://github.com/matter-labs/zksync-era/pull/2474/commits/3aa30b230794a2a02721abfce63ac5fc69579273). ✅